token/services/ttxdb/db/badger/badger.go

```diff
@@ -569,6 +570,81 @@ func (db *Persistence) GetTransactionEndorsementAcks(txID string) (map[string][]
 	return acks, nil
 }
 
+func (db *Persistence) StoreCertifications(certifications map[*token.ID][]byte) error {
+	txn := db.db.NewTransaction(true)
+	defer txn.Discard()
+
+	for tokenID, certification := range certifications {
+		key, err := db.certificationKey(tokenID)
+		if err != nil {
+			return errors.Wrapf(err, "could not get key for token id %s", tokenID)
+		}
+		logger.Debugf("adding certification [%s] with key [%s]", tokenID, key)
+
+		err = txn.Set([]byte(key), certification)
+		if err != nil {
+			return errors.Wrapf(err, "could not set value for key %s", key)
+		}
+	}
+	if err := txn.Commit(); err != nil {
+		return errors.Wrapf(err, "could not commit certifications")
+	}
+	return nil
+}
+
+func (db *Persistence) GetCertifications(ids []*token.ID, callback func(*token.ID, []byte) error) error {
```

Comment:
Maybe it would be more intuitive to return the results instead:
```
type Certification = []byte
GetCertifications(ids []*token.ID) ([]Certification, error)
```

Reply:
Good point. I have the callback function to let the caller to further marshal the certification to a data structure the caller understands.